### ISSUE TITLE
findAll is a static member

### ISSUE THREAD

**Issue by zaharzagrava** [sequelize/sequelize#12773]
### What are you doing?

Somewhere in my code I create a model that eventually has type `Model<Client, CreateClientDTO>`.

```js
// Somewhere in other file
class Client extends Model<Client, CreateClientDTO> {
   /*I provide all sorts of methods and later create an instance of this class*/
}

// In other or same file
import client from '../path_to_myClientInstance';

export default class SomeClass {
  static async getAll() {
    const instances = await client.findAll();
  }
}

```

### What do you expect to happen?

Expect `findAll` to work, but since `findAll` is a static member of Model type that resides in `types/lib/model.d.ts`, I cant call it on instance.

### What is actually happening?

Typescript thrown error: `"Property 'findAll' is a static member of type 'Client'ts(2576)"` and reasonably so.

### Environment

- Sequelize version: "^6.3.5"
- Node.js version: 12.19.0
- Operating System: Ubuntu 18.04.04 LTS
- If TypeScript related: TypeScript version: 4.0.3

### How does this problem relate to dialects?

- [+ ] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s)
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [+ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by jabin753** [comment]
Do you pass any sequelize instance to your class?
Personally I don't use static methods often, but I found this structure useful:
```typescript
export class ProductDao {
  constructor(public model: typeof ProductModel) {}
  /**
   * Sequelize wrapper Model.findAll
   */
  public getAll(): Promise<IProductData[]> {
    return this.model.findAll()
  }
```
Init:
```typescript

let DB: IDB | null
export async function provider(): Promise<IProvider> {
  if (!DB) DB = await setupDatabase()
  return {
    products: new ProductDao(DB.Product), //DB object is a set of initialize sequelize models
  }
}
```
Usage in your backend: 
```typescript
const products = await provider().products.getAll()
```

**Comment by lazcanocode** [comment]
I have the same problem, I'm trying of inject the model in my store class but it say me 
"Property 'findAll' is a static member of type 'Model<TModelAttributes, TCreationAttributes>'."  

```ts
import { Model } from 'sequelize';

export class Store<TModelAttributes, TCreationAttributes> {
  model: Model<TModelAttributes, TCreationAttributes>;

  constructor (model: Model<TModelAttributes, TCreationAttributes>) {
    this.model = model;
  }

  async getAll (): Promise<Model[]> {
    const data = await this.model.findAll();
    return data;
  }
}
```

**Comment by papb** [comment]
Hello, this is a common source of confusion. Note that if an object is of type `Model`, then it means it is an instance of that class. For example, if you have a `class User extends Model`, and then you do `const someone = await User.create({})`, then `someone` will have type `Model`. Note that `User` does not have type `Model`. In other words, `User` is not a `Model` from TS point of view. The type of `someone` is `Model`. The type of `User` is `typeof Model`, not `Model`.

For example, @lazcanocode:

```ts
import { Model } from 'sequelize';

export class Store {
  model: typeof Model;

  constructor (model: typeof Model) {
    this.model = model;
  }

  async getAll (): Promise<Model[]> {
    const data = await this.model.findAll();
    return data;
  }
}
```

**Comment by snyam2095** [comment]
Hi @papb,

I tried the method that you suggested but I ran into some problem.

I am attaching the screenshot of the error I am getting.

<img width="1440" alt="Screenshot 2024-05-21 at 18 03 53" src="https://github.com/sequelize/sequelize/assets/107630037/f590b3da-d231-4574-a585-041e7e36f314">


`import { Model } from 'sequelize';

export class BaseDao {
    private model: typeof Model;

    constructor(dbModel: typeof Model) {
        this.model = dbModel
    }


    async findAll(query: object): Promise<Model[]> {
        const result = await this.model.findAll(query)
        return result
    }

}`

this is the code I wrote. Would love to hear from you.
